app_legacy/Helpers/render/code-note.php

```diff
@@ -1,41 +1,64 @@
 <?php
 
-function RenderCodeNotes(array $codeNotes): void
+function RenderCodeNotes(array $codeNotes, bool $editable = false): void
 {
-    echo "<table class='table-highlight'><tbody>";
+    echo "<table class='table-highlight'>";
 
-    echo "<tr class='do-not-highlight'><th style='font-size:100%;'>Mem</th><th style='font-size:100%;'>Note</th><th style='font-size:100%;'>Author</th></tr>";
+    echo "<thead>";
+    echo "<tr class='do-not-highlight'>";
+    echo "<th style='font-size:100%;'>Mem</th>";
+    echo "<th style='font-size:100%;'>Note</th>";
+    echo "<th style='font-size:100%;'>Author</th>";
+    if ($editable) {
+        echo "<th>Dev</th>";
+    }
+    echo "</tr>";
+    echo "</thead>";
+
+    echo "<tbody>";
 
+    $rowIndex = 0;
     foreach ($codeNotes as $nextCodeNote) {
         if (empty(trim($nextCodeNote['Note'])) || $nextCodeNote['Note'] == "''") {
             continue;
         }
 
-        echo "<tr>";
+        echo "<tr id='row-$rowIndex'>";
 
         $addr = $nextCodeNote['Address'];
         $addrInt = hexdec($addr);
 
         $addrFormatted = sprintf("%04x", $addrInt);
-        $memNote = $nextCodeNote['Note'];
+        $originalMemNote = $nextCodeNote['Note'];
 
-        sanitize_outputs($memNote);
+        sanitize_outputs($originalMemNote);
 
-        $memNote = nl2br($memNote);
+        $memNote = nl2br($originalMemNote);
 
-        echo "<td style='width: 25%;'>";
+        echo "<td data-address='$addr' style='width: 25%;'>";
         echo "<span class='font-mono'>0x$addrFormatted</span>";
         echo "</td>";
 
         echo "<td>";
-        echo "<div class='font-mono' style='word-break:break-word'>$memNote</div>";
+        echo "<div class='font-mono note-display block' style='word-break:break-word'>$memNote</div>";
+        echo "<textarea class='w-full font-mono note-edit hidden'>$originalMemNote</textarea>";
+        echo "<button class='save-btn hidden' onclick='saveCodeNote($rowIndex)'>Save</button>";
         echo "</td>";
 
-        echo "<td>";
+        echo "<td class='note-author-avatar' data-current-author='" . $nextCodeNote['User'] . "'>";
         echo userAvatar($nextCodeNote['User'], label: false, iconSize: 24);
         echo "</td>";
 
+        if ($editable) {
+            echo "<td>";
+            echo "<button class='edit-btn inline' onclick='beginEditMode($rowIndex)'>Edit</button>";
```

Comment:
Good catch.

I have made some slight adjustments so that Jr Devs only see an Edit button by code notes they are the author of.

As far as Jr Dev edits disappearing, this turned out to be a bug in my _update-code-note.php_ implementation's `authenticateFromCookie` call.

I have addressed these issues in https://github.com/RetroAchievements/RAWeb/pull/1440/commits/3b017ef1ac1b059a4ccb62a584bf8a93ced8e4b1.